src/vm/contexts.rs

```diff
@@ -29,6 +31,17 @@ pub struct OwnedEnvironment <'a> {
     call_stack: CallStack
 }
 
+/**
+ The AssetMap is used to track which assets have been transfered from whom
+ during the execution of a transaction.
+ */
+#[derive(Debug)]
+pub struct AssetMap {
+    // Q: currently we just track balance transfers, but for NFT,
+    //     tracking the actual identifier transfered is probably more useful.
+    map: HashMap<PrincipalData, HashMap<AssetIdentifier, i128>>
```

Comment:
We should probably add NFTs here as well.  You could imagine a single transaction that, when evaluated, causes an asset to change hands through multiple owners during execution (i.e. as a series of `contract-call!`s).

Reply:
Yeah -- that's probably the right idea -- currently NFTs just get logged as a "transfer of 1", but we should specifically enumerate the NFTs.

Reply:
Addressed in 1cf8cf40d231e8dfe1f3f43c149b9a40107ab064